pkg/iam/iam.go

```diff
@@ -608,3 +606,11 @@ func roleName(role string, clusterID string) string {
 func policyName(role string, clusterID string) string {
 	return fmt.Sprintf("%s-%s-policy", role, clusterID)
 }
+
+func getServiceAccount(role string) string {
+	if role == CertManagerRole {
+		return "cert-manager-controller"
+	}
+
+	return "external-dns"
```

Comment:
This seems dangerous. Let's instead return `(string, error)` and fail if the role value isn't supported.